Feature - set ARM outputs to Azure DevOps variable group (#94)
arcus-azure/arcus.scripting: docs/preview/features/powershell/azure-devops.md

Assign a value to a DevOps pipeline variable during the execution of this pipeline.

| Parameter       | Mandatory | Description                                       |
| --------------- | --------- | ------------------------------------------------- |
| `Name`          | yes       | The name of the variable to set in the pipeline   |
| `Value`         | yes       | The value of the variable to set in the pipeline  |

**Example**

```powershell
PS> Set-AzDevOpsVariable "my-variable" "my-variable-value"
# #vso[task.setvariable variable=my-variable] my-variable-value
```

## Setting ARM outputs to Azure DevOps variable group

> Sets the Azure Resource Management (ARM) outputs as a variable group on Azure DevOps.

Reviewer: ```suggestion
Stores the Azure Resource Management (ARM) outputs in a variable group on Azure DevOps.
```

| Parameter                      | Mandatory | Description                                                                                             |
| ------------------------------ | --------- | ------------------------------------------------------------------------------------------------------- |

> | `VariableGroupName`            | yes       | The name of the variable group on Azure DevOps where the ARM outputs should be set                      |

Reviewer: ```suggestion
| `VariableGroupName`            | yes       | The name of the variable group on Azure DevOps where the ARM outputs should be stored                      |
```

| `UpdateVariablesForCurrentJob` | no        | The switch to also set the variables in the ARM output as pipeline variables in the current running job |

**Example**
Without updating the variables in the current job running the pipeline:

```powershell
PS> Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName "my-variable-group"
# Adding variable $output.Name with value $variableValue to variable group my-variable-group
# Retrieving project details
# Set properties for update of existing variable group
```

> With also updating the variables in the current job running the pipeline:

Reviewer: ```suggestion
Include updating the variables in the current job running the pipeline, to immediately make them available to the next pipeline tasks:
```

```powershell

> PS> Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName "my-variable-group"

Reviewer: The switch itself is missing here:
```suggestion
PS> Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName "my-variable-group" -UpdateVariablesForCurrentJob
```

Author: Thanks!

**Example**
Without updating the variables in the current job running the pipeline:

```powershell
PS> Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName "my-variable-group"
# Get ARM outputs from 'ArmOutputs' environment variable
# Adding variable $output.Name with value $variableValue to variable group my-variable-group
# Retrieving project details
# Set properties for update of existing variable group
```

Include updating the variables in the current job running the pipeline, to immediately make them available to the next pipeline tasks:

```powershell
PS> Set-AzDevOpsArmOutputsToVariableGroup -VariableGroupName "my-variable-group" -UpdateVariablesForCurrentJob
# Get ARM outputs from 'ArmOutputs' environment variable
# Adding variable $output.Name with value $variableValue to variable group my-variable-group
# Retrieving project details
# Set properties for update of existing variable group
# The pipeline variable $variableName will be updated to value $variableValue as well, so it can be used in subsequent tasks of the current job. 
# ##vso[task.setvariable variable=$variableName]$variableValue
```

> Include custom environment variable for the ARM outputs:

Reviewer: ```suggestion
Include user-defined environment variable for the ARM outputs:
```
